documentation/api/query/tutorial-pql.markdown

```diff
@@ -0,0 +1,403 @@
+not numeric, and numeric comparison operators can't be used with them at
+this time.
+
+## Nodes walkthrough
+
+We can also query for nodes. Once again, this is quite similar to resource and
+fact queries:
+
+    nodes {
+      certname = "foo.example.com"
+    }
+
+The result of this query is:
+
+    [ {
+        "deactivated" : null,
+        "facts_environment" : "production",
+        "report_environment" : "production",
+        "catalog_environment" : "production",
+        "facts_timestamp" : "2015-06-22T17:25:11.886Z",
+        "expired" : null,
+        "report_timestamp" : "2015-06-22T17:25:07.484Z",
+        "certname" : "foo.example.com",
+        "catalog_timestamp" : "2015-06-22T17:25:12.023Z",
+        "latest_report_hash" : "754b0b87af9ee647507b5aa3001f44f8e8843216",
+        "latest_report_status" : "unchanged"
+    } ]
+
+This will return an object containing the certname "foo.example.com", as well
+as some metadata detailing deactivation status and the most recent fact,
+report, and catalog updates from that node.
+
+## Subquery walkthrough
+
+### Explicit subqueries
+
+The queries we've looked at so far are quite powerful and useful, but what if
+your query needs to consider both resources *and* facts?
+
+For instance, suppose you're configuring a load balancer, and need the IP addresses of your Apache servers. You could find those servers by using this resource query:
+
+    resources {
+      type = "Class" and
+      title = "apache"
+    }
+
+This will find all the Class[Apache] resources, which each knows the certname
+of the node it came from. Then you could put all those certnames into a fact
+query:
+
+    facts {
+      name = "ipaddress" and
+      (certname = "a.example.com" or
+       certname = "b.example.com" or
+       certname = "c.example.com" or
+       certname = "d.example.com" or
+       certname = "e.example.com")
+    }
+
+But this query is lengthy, and it requires some logic to assemble and run the
+second query. There has to be a better way! What if we could find the
+Class[Apache] servers and use the results of that query to find the
+certname? We can, with this fact query:
+
+    facts {
+      name = "ipaddress" and
+      certname in resources[certname] { type = "Class" and title = "apache" }
+    }
+
+This may appear a little daunting, so we'll look at it piece by piece.
+
+Let's start with `resources[certname] { <filter> }`. This query would return the
+certname for the results that match the filter as specified.
+
+We then use an operator called "in" to turn our list of resources into
+just a list of certnames. So we now conceptually have something like:
+
+    certname in ["foo.example.com", "bar.example.com", "baz.example.com"]
+
+The "in" operator matches facts whose "certname" is in the supplied list. (In our case, this list is generated by a subquery. To use a literal list, you must use the the "list" syntax described in the [PQL lists][lists].)
+
+### Implicit subqueries
```

Comment:
Might be nice to have links to the more thorough documentation for each section in here :-)